homeassistant/components/squeezebox/media_player.py

```diff
@@ -65,52 +71,61 @@
     | SUPPORT_CLEAR_PLAYLIST
 )
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_USERNAME): cv.string,
-    }
-)
-
-DATA_SQUEEZEBOX = "squeezebox"
-
-KNOWN_SERVERS = "squeezebox_known_servers"
-
+KNOWN_PLAYERS = "known_players"
+KNOWN_SERVERS = "known_servers"
+DISCOVERY_TASK = "discovery_task"
 ATTR_PARAMETERS = "parameters"
-
 ATTR_OTHER_PLAYER = "other_player"
 
 ATTR_TO_PROPERTY = [
     ATTR_QUERY_RESULT,
     ATTR_SYNC_GROUP,
 ]
 
+SQUEEZEBOX_MODE = {
+    "pause": STATE_PAUSED,
+    "play": STATE_PLAYING,
+    "stop": STATE_IDLE,
+}
+
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the squeezebox platform."""
+    """Set up squeezebox platform from platform entry in configuration.yaml (deprecated)."""
+    _LOGGER.warning("Loading Squeezebox via platform config is deprecated.")
+
+    if config:
+        conf = {
+            CONF_HOST: config.get(CONF_HOST),
+            CONF_PORT: config.get(CONF_PORT) if config.get(CONF_PORT) else DEFAULT_PORT,
+        }
+        for attr in [CONF_USERNAME, CONF_PASSWORD]:
+            if attr in config:
+                conf.update(attr, config.get(attr))
+        await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=conf
+        )
 
-    known_servers = hass.data.get(KNOWN_SERVERS)
-    if known_servers is None:
-        hass.data[KNOWN_SERVERS] = known_servers = set()
 
-    if DATA_SQUEEZEBOX not in hass.data:
-        hass.data[DATA_SQUEEZEBOX] = []
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up an LMS Server from a config entry."""
+    config = config_entry.data
+    _LOGGER.debug("Reached async_setup_entry, config=%s", config)
 
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
```

Comment:
```suggestion
    host = config[CONF_HOST]
```

Host is required right?

Reply:
Yes.